src/zenml/zen_server/routers/runs_endpoints.py

```diff
@@ -71,6 +72,28 @@
 logger = get_logger(__name__)
 
 
+def check_concurrent_run_limit() -> None:
+    """Check if the concurrent run limit is exceeded.
+
+    Raises:
+        SubscriptionUpgradeRequiredError: If the limit of concurrent runs is
+            exceeded.
+    """
+    max_concurrent_runs = server_config().max_concurrent_pipeline_runs
+
+    if (
+        max_concurrent_runs
+        and zen_store().count_active_runs() >= max_concurrent_runs
+    ):
+        raise SubscriptionUpgradeRequiredError(
+            "You have reached the maximum number of concurrent runs. "
+            "Please upgrade your subscription to increase the limit. "
+            "If some of your runs are incorrectly still in running state even "
+            "though they've failed already, you can manually delete them by "
+            "running `zenml pipeline runs delete <ID>`."
```

Comment:
so there is no way to fetch the active run id or something to make this easier?

Reply:
There is no "the active run", this is a generic function and not all servers are having just a single concurrent run as limit. Even so, I would never give them the command here as they will just blindly execute it anyway, this way they have to actually look for the run and actively decide which run to delete.